cardinal/ecs/persona.go

```diff
@@ -133,40 +146,55 @@ func RegisterPersonaSystem(wCtx WorldContext) error {
 	if err != nil {
 		return err
 	}
-	for _, txData := range createTxs {
-		tx := txData.Msg
-		if _, ok := personaTagToAddress[tx.PersonaTag]; ok {
+
+	CreatePersonaMsg.Each(wCtx, func(txData TxData[CreatePersona]) (CreatePersonaResult, error) {
+		msg := txData.Msg
+		result := CreatePersonaResult{Success: false}
+
+		if valid := isAlphanumericWithUnderscore(msg.PersonaTag); !valid {
```

Comment:
```suggestion
		if !isAlphanumericWithUnderscore(msg.PersonaTag) {
```

Reply:
can prob just create a persona normalization and validator function in /types/persona